sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java

```diff
@@ -56,38 +48,26 @@ public static SdkMeterProviderBuilder builder() {
   }
 
   SdkMeterProvider(
-      List<MetricReader> metricReaders,
+      List<RegisteredReader> registeredReaders,
       Clock clock,
       Resource resource,
       ViewRegistry viewRegistry,
-      ExemplarFilter exemplarFilter,
-      long minimumCollectionIntervalNanos) {
+      ExemplarFilter exemplarFilter) {
     this.sharedState =
         MeterProviderSharedState.create(clock, resource, viewRegistry, exemplarFilter);
+    this.registeredReaders = registeredReaders;
+    for (RegisteredReader registeredReader : registeredReaders) {
+      registeredReader.getReader().register(new LeasedMetricProducer(registeredReader));
```

Comment:
Compared to the previous code, it became less obvious why we would call this seemingly circular registration. Maybe a bit clearer to make `LeasedMetricProducer` static and pass in what it uses (or even `this`, just to make it explicit what state is being registered)